addons/link_tracker/tests/test_mail_render_mixin.py

```diff
@@ -104,8 +102,52 @@ def test_shorten_links_html_including_base_url(self):
         matches = expected_pattern.search(new_content).groups()
         # 3rd and 4th lead to the same short_url
         self.assertEqual(matches[0], matches[1])
-        # 5th has different label but should currently lead to the same link
-        self.assertEqual(matches[1], matches[2])
+
+    def test_shorten_links_html_different_labels(self):
+        content = """
+There is a <a href="https://www.odoo.com">link</a> here,
+<a href="https://www.odoo.com">there</a>, and in this
+<a href="https://www.odoo.com"><img src="https://www.odoo.com/logo.png" alt="image"/></a>
+and also <a href="https://www.odoo.com">
+    <img src="https://www.odoo.com/logo.png" fakealt="image3" alt="image2's trouble"></img>
+</a>
+Single/Nested quotes are not <a href="https://www.odoo.com"><img src='https://www.odoo.com/logo.png' alt='"scary"'/></a>
+Nor escaped <a href="https://www.odoo.com"> blurp <img src="https://www.odoo.com/logo.png" alt="ins \' ide"></a>
+        """.format(base_url=self.base_url)
+
+        expected_pattern = re.compile("""
```

Comment:
here you can use `r` and remove double escaping
```suggestion
        expected_pattern = re.compile(r"""
```

Reply:
I applied this in these tests and updated all to use it (and f-formatting), which makes a slightly bigger diff but does reduce the final amount of code :-)